packages/main/src/plugin/telemetry/telemetry.ts

```diff
@@ -92,59 +91,22 @@ export class Telemetry {
     this.configurationRegistry.registerConfigurations([telemetryConfigurationNode]);
 
     // grab value
-    const telemetryConfiguration = this.configurationRegistry.getConfiguration('telemetry');
-    const check = telemetryConfiguration.get<boolean>('check');
+    const telemetryConfiguration = this.configurationRegistry.getConfiguration(TelemetrySettings.SectionName);
+    const check = telemetryConfiguration.get<boolean>(TelemetrySettings.Check);
 
     // initalize objects
     this.analytics = new Analytics(Telemetry.SEGMENT_KEY);
 
     // needs to prompt the user for the first time he launches the app
-    if (!check) {
-      dialog
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        .showMessageBox(findWindow()!, {
-          title: 'Telemetry',
-          message: `Help Red Hat improve\n ${app.getName()} by allowing anonymous usage data to be collected.`,
-          buttons: ['OK'],
-          type: 'info',
-          checkboxChecked: true,
-          checkboxLabel: 'Enable telemetry',
-          detail:
-            'Read about our privacy statement:\n https://developers.redhat.com/article/tool-data-collection\n\nYou can update your choice later by changing the Telemetry preference.',
-        })
-        .then(async messageBoxReturnValue => {
-          // check has been performed, we asked the user and he answered, so check is done
-          await this.configurationRegistry.updateConfigurationValue(
-            'telemetry.check',
-            true,
-            CONFIGURATION_DEFAULT_SCOPE,
-          );
-          // if the user said yes, we enable the telemetry
-          if (messageBoxReturnValue.checkboxChecked) {
-            await this.configurationRegistry.updateConfigurationValue(
-              'telemetry.enabled',
-              true,
-              CONFIGURATION_DEFAULT_SCOPE,
-            );
-            this.configureTelemetry();
-          } else {
-            // else we disable it
-            await this.configurationRegistry.updateConfigurationValue(
-              'telemetry.enabled',
-              false,
-              CONFIGURATION_DEFAULT_SCOPE,
-            );
-          }
-        });
-    } else {
-      //
-      const enabled = telemetryConfiguration.get<boolean>('enabled');
-      if (enabled === true) {
-        await this.configureTelemetry();
-        this.telemetryConfigured = true;
-        this.telemetryEnabled = true;
+    if (check) {
+      const enabled = telemetryConfiguration.get<boolean>(TelemetrySettings.Enabled);
+      if (enabled === true) await this.configureTelemetry();
+      else {
```

Comment:
need to check why formatter/linter didn't complained there

```suggestion
      if (enabled === true) {
        await this.configureTelemetry();
      } else {
```

Reply:
Fixed in second commit.